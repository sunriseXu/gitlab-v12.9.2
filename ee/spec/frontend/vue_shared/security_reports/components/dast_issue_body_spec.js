import { shallowMount } from '@vue/test-utils';
import DastIssueBody from 'ee/vue_shared/security_reports/components/dast_issue_body.vue';

describe('Dast Issue Body', () => {
  let wrapper;

  const createComponent = () => {
    wrapper = shallowMount(DastIssueBody, {
      propsData: {
        issue: {
          alert: 'X-Content-Type-Options Header Missing',
          severity: 'Low',
          count: '17',
          cweid: '16',
          desc:
            '<p>The Anti-MIME-Sniffing header X-Content-Type-Options was not set to "nosniff". </p>',
          title: 'X-Content-Type-Options Header Missing',
          reference:
            '<p>http://msdn.microsoft.com/en-us/library/ie/gg622941%28v=vs.85%29.aspx</p><p>https://www.owasp.org/index.php/List_of_useful_HTTP_headers</p>',
          riskcode: '1',
          riskdesc: 'Low (Medium)',
        },
        status: 'failed',
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('matches the snaphot', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });
});
