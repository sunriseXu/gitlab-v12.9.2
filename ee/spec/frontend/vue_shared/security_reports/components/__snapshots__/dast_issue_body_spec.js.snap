// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dast Issue Body matches the snaphot 1`] = `
<div
  class="report-block-list-issue-description prepend-top-5 append-bottom-5"
>
  <div
    class="report-block-list-issue-description-text"
  >
    <severity-badge-stub
      class="d-inline-block"
      severity="Low"
    />
     
    <modal-open-name-stub
      class="js-modal-dast"
      issue="[object Object]"
      status="failed"
    />
  </div>
</div>
`;
